.github/workflows/release.yml

```diff
@@ -8,26 +8,14 @@ on:
 jobs:
   release-pypi:
     if: github.repository_owner == 'gammapy'
-    runs-on: ubuntu-latest
-    steps:
-      - name: Checkout
-        uses: actions/checkout@v4
-      - name: Update tags
-        run: git fetch --tags --force
-      - name: Set up Python
-        uses: actions/setup-python@v5
-        with:
-          python-version: 3.9
-      - name: Install dependencies
-        run: |
-          python --version
-          pip install -U build
-          python -m build --sdist
-      - name: Publish package
-        uses: pypa/gh-action-pypi-publish@release/v1
-        with:
-          user: __token__
-          password: ${{ secrets.PYPI_TOKEN }}
+    uses: OpenAstronomy/github-actions-workflows/.github/workflows/publish.yml@v1
+    with:
+      targets: |
+        - cp3*-manylinux_x86_64*
+        - cp3*-macosx*
+        - cp3*-win_amd64*
+    secrets:
+      pypi_token: ${{ secrets.PYPI_TOKEN }}
```

Comment:
Does this GitHub Action not use [trusted publishers](https://docs.pypi.org/trusted-publishers/) yet? I'm in general not a fan of these "build and publish all in one go" workflows as they usually end up causing a lot of pain when they mess up. It is better to build your distributions in a different step from publishing. It also reduces the permissions that you need to give to any one workflow, which is better for security.

Reply:
No, I think we haven't configured trusted publishers yet, I'll tag @registerrier on this. I agree on the separate build and publishing step. Alternatively we could either cron or independently trigger the builds for testing as described in the link. This way we can still keep the workflow @maxnoe proposed.

Reply:
The pipeline is actually multi stage, this is not a single action building and publishing wheels, see:
https://github.com/gammapy/gammapy/actions/runs/13416279534

I thought it makes sense to re-use the open astronomy workflow, which is e.g. also used by astropy:

https://github.com/astropy/astropy/blob/60aa54bc65f59f97c02b26d7c00063a3fa5a52a1/.github/workflows/publish.yml#L23-L30

But if you prepare to roll out a custom workflow or one based on another upstream resource, fine with me

Reply:
Thanks @maxnoe for clarifying! I agree we should re-use existing workflows from the astronomy community. I guess the difference to Astropy is that they have configured nightly builds? This way they always see, when builds fail and can fix the next day. While for us it will only trigger on a release tag. But as we usually do release candidates I am not too worried, so we could still fix things...